src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py

```diff
@@ -26,5 +28,27 @@ def load_arguments(self, _):
 
     with self.argument_context('consumption pricesheet show') as cps:
+        cps.argument('top', options_list=['--top', '-t'], type=int, help='maximum number of items to return. Accepted range for this value is 1 - 1000')
         cps.argument('include_meter_details', options_list=['--include-meter-details', '-m'], action='store_true', help='include meter details in the price sheet')
         cps.argument('billing_period_name', options_list=['--billing-period-name', '-p'], help='name of a specific billing period to get the price sheet')
+
+    with self.argument_context('consumption marketplace list') as cmp:
+        cmp.argument('billing_period_name', options_list=['--billing-period-name', '-p'], help='name of a specific billing period to get the marketplace')
+        cmp.argument('top', options_list=['--top', '-t'], type=int, help='maximum number of items to return. Accepted range for this value is 1 - 1000')
+        cmp.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='start date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
+        cmp.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='end date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
+
+    with self.argument_context('consumption budget') as cb:
+        cb.argument('resource_group_name', options_list=['--resource-group-name', '-r'], action='store_true', help='get budget list for subscription by specific resource group name')
+        cb.argument('budget_name', options_list=['--budget-name', '-b'], help='get budget information by budget name')
+        cb.argument('category', options_list=['--category', '-c'], type=str, help='create budget with category')
+        cb.argument('amount', options_list=['--amount', '-a'], type=get_decimal_type(), help='create budget with amount')
+        cb.argument('time_grain', options_list=['--time_grain', '-t'], type=str, help='create budget with time grain')
+        cb.argument('start_date', options_list=['--start_date', '-s'], type=get_datetime_type(), help='create budget with start date of time period')
+        cb.argument('end_date', options_list=['--end_date', '-e'], type=get_datetime_type(), help='create budget with end date of time period')
+        cb.argument('resource_groups', options_list=['--resource-groups', '-g'], nargs='+', help='update budget with resource groups')
+        cb.argument('resources', options_list=['--resources', '-u'], nargs='+', help='update budget with resource(s) specified')
+        cb.argument('meters', options_list=['--meters', '-m'], nargs='+', help='update budget with meter(s) specified')
+
+    with self.argument_context('consumption budget update') as cb:
+        cb.argument('e_tag', options_list=['--e-tag', '-x'], type=str, help='etag required when updating existing budget')
```

Comment:
`-x` is not a sensible short option. "required when updating existing budget".  How can you update a non-existing budget?  The update command needs to conform to the generic update CLI command pattern. The e-tag can be used from the retrieved instance, making it unnecessary to even expose this parameter.  
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#generic-update-commands